server/src/main/java/org/elasticsearch/index/mapper/IgnoredSourceFieldMapper.java

```diff
@@ -170,7 +170,7 @@ static byte[] encode(NameValue values) {
 
         byte[] nameBytes = values.name.getBytes(StandardCharsets.UTF_8);
         byte[] bytes = new byte[4 + nameBytes.length + values.value.length];
-        ByteUtils.writeIntLE(values.name.length() + PARENT_OFFSET_IN_NAME_OFFSET * values.parentOffset, bytes, 0);
+        ByteUtils.writeIntLE(nameBytes.length + PARENT_OFFSET_IN_NAME_OFFSET * values.parentOffset, bytes, 0);
```

Comment:
Just double checking, there is no need for an index version check here, given that decode isn't updated in this change. In other words, Indexing new documents in indices with older index version, would result in the error described in the PR description to not occur.